### ISSUE TITLE
[BUG] Noticeable logits shifting to large value

### ISSUE THREAD

**Issue by siriusctrl** [microsoft/DeepSpeed#4195]
**Describe the bug**
Over the past few days, we conducted an update and observed an interesting pattern related to the issue of logit increase. It appears that when the GPU load during the training process is relatively high, the problem of logit increase does not manifest. Conversely, when the GPU load is lower, we consistently encounter the logit increase problem. To illustrate this with an example, when using a batch size of 4 million and after gone though a fixed amount of tokens.

1. training on 184 A100 GPUs, the issue becomes apparent.
2. However, by reducing the number of GPUs to 144 and subsequently increasing the load on each individual GPU, the problem becomes not so obvious
3. By further increasing the number of GPUs to 256, we found the logits increase much faster (than 184 cards) 
4. By increase the global size from 4 million to 16 million, the problem becomes not so obvious again for 256 cards.

This problem persist with Transformers (with deepspeed) and megatron-deepspeed(zero1 from deepspeed), but seems to disappear while using only Megatron. Thus, we guess this might be related to deepspeed.

**To Reproduce**
We trained models following the setup of Open-llama with architecture the same as Llama. The data ratio can be found here https://github.com/openlm-research/open_llama/issues/84

**Comment by TissueC** [comment]
Hi, I encountered the same issue recently. I think this is really serious and noticeable because high logits could greatly injure the performance though the loss (or softmax output) seems normal. May I ask you how you solved this problem? And do you know the threshold of "high" and "low" GPU load, which could help avoid this problem.
